### DISCUSSION TITLE
Service to Show A Hotspot QR Code on boot

### DISCUSSION THREAD

**Discussion by DanielLester83** [RaspAP/raspap-webgui#1219] [Ideas]
Required:
Install the framebuffer image program
`sudo apt-get install fbi`

Create the Service file located at /etc/systemd/system/splashscreen.service

```
[Unit]
Description=Splash screen
DefaultDependencies=no
After=local-fs.target

[Service]
ExecStart=/bin/sh -cv "sed -n 's/^ssid=\(.*\)/WIFI:T:WPA;S:\1;P:/p;s/^wpa_passphrase=\(.*\)/\1;;/p' < /etc/hostapd/hostapd.conf | tr -d '\n' | sudo /usr/bin/qrencode -m 2 -t utf8 -t PNG -o /dev/shm/splash.png && /usr/bin/fbi -d /dev/fb0 --noverbose -a /dev/shm/splash.png"
StandardInput=tty
StandardOutput=tty

[Install]
WantedBy=sysinit.target
StandardInput=tty
StandardOutput=tty

[Install]
 #WantedBy=sysinit.target
```

Enable the service
`sudo systemctl enable splashscreen.service`

Suggested:
Disable the Boot Splash Image
add this to /boot/config.txt
`disable splash=1`

Disable most Boot Text
add this to /boot/cmdline.txt
` logo.nologo consoleblank=0 loglevel=1 quiet`

Disable the command line prompts
`sudo systemctl disable getty@tty*`

**Comment by billz** [comment]
Nice! (Minor edit to the 'Unit' stanza)
